mdsuite/file_io/extxyz_files.py

```diff
@@ -0,0 +1,305 @@
+"""
+MDSuite: A Zincwarecode package.
+
+License
+-------
+This program and the accompanying materials are made available under the terms
+of the Eclipse Public License v2.0 which accompanies this distribution, and is
+available at https://www.eclipse.org/legal/epl-v20.html
+
+SPDX-License-Identifier: EPL-2.0
+
+Copyright Contributors to the Zincwarecode Project.
+
+Contact Information
+-------------------
+email: zincwarecode@gmail.com
+github: https://github.com/zincware
+web: https://zincwarecode.com/
+
+Citation
+--------
+If you use this module please cite us with:
+
+Summary
+-------
+"""
+import logging
+import typing
+from typing import Union, List, Dict, Tuple
+import numpy as np
+import pathlib
+import copy
+import tqdm
+
+import mdsuite.file_io.file_read
+import mdsuite.database.simulation_database
+from mdsuite.utils.meta_functions import get_dimensionality
+
+log = logging.getLogger(__name__)
+
+var_names = {
+    "Positions": "pos",
+    "Velocities": "vel",
+    "Forces": "force",
+    "Stress": "stress",
+    "PE": "energies",
+    "Time": "time",
+    "Lattice": "Lattice",
+    "Momenta": "momenta",
+}
+
+
+class EXTXYZFile(mdsuite.file_io.file_read.FileProcessor):
+    """
+    Reader for extxyz files
+    """
+
+    def __init__(self, file_path: str, custom_data_map: dict = None):
+        self.file_path = pathlib.Path(file_path).resolve()
+        self.n_header_lines = 2
+        if custom_data_map is None:
+            custom_data_map = {}
+        self.custom_data_map = custom_data_map
+
+        self._n_particles = None
+        self._mdata = None
+        self._properties_dict = None
+        self._species_dict = None
+        self._batch_size = None
+
+    def __str__(self):
+        return str(self.file_path)
```

Comment:
I think this can be in the parent class, because you use the same method in every file reader. If a filereader needs adjustments it can be overwritten

Reply:
the script input reader does not have a file path.

Reply:
maybe wrap it in a try..except then? It feels a little weird to have this duplicated in every class?

Reply:
my point was to force the reader dev to think about what makes the file reader instance unique

Reply:
It is not used anywhere in the code, only for representation to the user I guess, so forcing the develeper to implent it is a good idea.

Reply:
it is used in the 'files already read'-check of the experiment. It is now in the tabular text parent class